client/src/components/ActiveChat/ActiveChat.js

```diff
@@ -23,7 +24,23 @@ const useStyles = makeStyles(() => ({
 const ActiveChat = (props) => {
   const classes = useStyles();
   const { user } = props;
-  const conversation = props.conversation || {};
+  const conversation = useMemo(() => props.conversation || {}, [props]);
+
+  const updateLastRead = useCallback(() => {
+    // get all messages received by user
+    const receivedMessages = conversation.messages.filter(mes => mes.senderId !== user.id);
+    // if user has received message, update last read message and update user
+    if(receivedMessages.length > 0){
+      const getLastReceivedMessage = receivedMessages.reduce((max, message) => (max.id > message.id) ? max : message);
+      props.updateLastReadMessages({userId: user.id, lastReads: conversation.lastReads, message: getLastReceivedMessage });
+    }
+  },[ props, conversation, user ]);
```

Comment:
Even though this is wrapped in a `useCallback()` this logic will be called every time that `props` change. That means every time that the component re-renders. This is very inefficient and the logic of retrieving the last read message code could be moved somewhere else, outside the component level.